src/pfe/file-watcher/server/test/unit-test/tests/projectUtil.module.test.ts

```diff
@@ -0,0 +1,482 @@
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+import { expect } from "chai";
+import * as path from "path";
+import * as fs from "fs";
+
+import * as crypto from "crypto";
+import { TransformOptions } from "stream";
+import * as projectUtil from "../../../../server/src/projects/projectUtil";
+import * as logHelperModule from "../lib/logHelper.module";
+import * as logHelper from "../../../../server/src/projects/logHelper";
+import * as libertyProject from "../../../../server/src/projects/libertyProject";
+import { existsAsync, mkdirAsync, writeAsync, copyAsync } from "../../functional-test/lib/utils";
+import * as projectsController from "../../../../server/src/controllers/projectsController";
+import { projectConstants } from "../../../../server/src/projects/constants";
+import * as app_configs from "../../functional-test/configs/app.config";
+import * as constants from "../../../../server/src/projects/constants";
+import { threadId } from "worker_threads";
+
+export function projectUtilTestModule(): void {
+    const extensionIDDir = path.join(process.env.CW_EXTENSION_DIR, "extensionProject");
+    it("test of getLogName function", async () => {
+        const projectID = "testProjectID";
+        const projectLocation = "directory/testproject";
+        const hash = crypto.createHash("sha1", <TransformOptions>"utf8").update(projectLocation);
+
+        const expectedResult  = projectConstants.containerPrefix + projectID + "-" + hash.digest("hex");
+        const actualResult = await projectUtil.getLogName(projectID, projectLocation);
+        expect(actualResult).to.equal(expectedResult);
+    });
+
+    it("test of getLogName function on Kube", () => {
+        const projectID = "testProjectID";
+        const projectLocation = "directory/testproject";
+        const hash = crypto.createHash("sha1", <TransformOptions>"utf8").update(projectLocation);
+        let expectedResult  = projectConstants.containerPrefix + projectID + "-" + hash.digest("hex");
+        if (expectedResult.length > 53) {
+            expectedResult = expectedResult.substring(0, 53);
+        }
+        process.env.IN_K8 = "true";
+        const actualResult = projectUtil.getLogName(projectID, projectLocation);
+        expect(actualResult).to.equal(expectedResult);
+        process.env.IN_K8 = "false";
+    });
+
+    describe("combinational testing of of getContainerName function", async () => {
+        before("create the extension directory with some fake files", async () => {
+            if (!(await existsAsync(extensionIDDir))) {
+                await mkdirAsync(extensionIDDir, { recursive: true });
+            }
+            expect(fs.statSync(extensionIDDir)).to.exist;
+            const filePath = path.resolve(extensionIDDir, ".sh-extension");
+            await writeAsync(filePath, '{"container": {"prefix": "testprefix-", "suffix": "-testsuffix"}}');
```

Comment:
Thx very much for the idea, I put it in my commit https://github.com/eclipse/codewind/pull/96/commits/f285533efb3ca6354a9c98be35d1d5932894d376